apps/chat_server/lib/chat_server/contexts/starred_messages/delete_starred_message.ex

```diff
@@ -0,0 +1,19 @@
+defmodule ChatServer.DeleteStarredMessage do
+  require Logger
+
+  alias ChatServer.Schema
+
+  def call(message_id, user) do
+    message = Schema.Message.get(message_id)
+    starred_message = Schema.StarredMessage.find_by_message_and_user(message, user)
+    star_id = Map.get(starred_message, :id)
+
+    with {:ok, _record} <- Schema.StarredMessage.delete(starred_message) do
```

Comment:
If only one clause is being evaluated, a `case` statement is more clear and concise.

The value of a `with` statement is it evaluates multiple clauses and guarantees they have returned the expected values.

In this instance, the function is making multiple calls. It could be rewritten where each is a `with` clause:

```elixir
def call(message_id, user) do
  with message <- Schema.Message.get(message_id),
       starred_message <- Schema.StarredMessage.find_by_message_and_user(message, user),
       {:ok, _record} <- Schema.StarredMessage.delete(starred_message) do
    {:ok, %{id: starred_message.id}}
  else
    # ...
  end
end
```